system/Model.php

```diff
@@ -330,7 +336,7 @@ protected function doInsertBatch(?array $set = null, ?bool $escape = null, int $
 	protected function doUpdate($id = null, $data = null, ?bool $escape = null): bool
 	{
 		$escape       = $this->escape;
-		$this->escape = null;
+		$this->escape = [];
```

Comment:
BaseModel Model->insert() 

Public methods like this handles Object (Entity) to array conversion ,validation, protect fields, timestamp processing for created_at, updated_at, deleted_at and triggers the events,

Model doInsert() - Also defined as abstract in BaseModel
 
The actual database connection to read, insert, update data happens in methods prefixed with `do` which are called at the specific point from mentioned public methods

What's the reason behind all of this. If you wan't to create model for any other kind of database which was in my example MongoDB. You previously had to override or implement everything in the custom model. Now you can simply extend the BaseModel and do the custom db implementation by only implementing the abstract methods from the BaseModel. In practice this means that all the non DB related features will be available in the custom model including validation, protect fields, timestamps, conversion to entity objects, soft deletes, callbacks, pager ....

It might seam easier to complete model from scratch but this saves a lot of time by reusing already well documented and tested functionalities especially when you follow the complete DB driver structure using ConnectionInterface, QueryInterface and ResultInterface

Reply:
@iRedds Thanks for the tip about the escape parameter I missed this one in #4072 so I submitted a new PR #4090 

@michalsn can you check it out and rebase this one afterward ?

Reply:
For example, I want to change the BaseModel::update() method.
In this case, I need to inherit the new class from BaseModel and implement all abstract methods. Instead of just extending the Model with existing methods.

Reply:
If you need to change the extend the update for SQL model you need to extend Model classs. Then you can either override the update or doUpdate depending on what you need

Extending BaseModel directly is only to be used when implementing Model class for custom database. eg. MongoDB, ElastecSearch, Firebase ...

Reply:
Thanks, @iRedds.

Thanks for making a PR @najdanovicivan, I missed it too. Please let me know if there are any issues with the proposed changes or if you see any place for improvements.